### PR TITLE
`Development`: Enhance coverage report script

### PR DESCRIPTION
**Pull request by N0W0RK**
#### General

- [x] This is a small issue that I tested locally.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/development-process.html#naming-conventions-for-github-pull-requests).

### Motivation and Context

The script for generating coverage reports was somewhat cumbersome to use.

### Description

- Better repo detection
  - The script can now be executed in any sub directory of the repo.
  - The pat to the repo can be specified via the `--repo-path` option.
 - Linux support
   - Updated README for Linux users
   - Enhanced error handling for pyperclip 

### Steps for Testing

Execute the script at `supporting_scripts/generate_code_cov_table/generate_code_cov_table.py` at different working directives inside the repo.
Execute the script from outside the repo specifying a path to the repo with the `--repo-path` option.

#### Code Review
- [ ] Code Review 1
- [ ] Code Review 2
#### Manual Tests
- [ ] Test 1
- [ ] Test 2

## Summary by CodeRabbit

- **Documentation**
	- Enhanced `README.md` with clearer instructions, including a new "Printing" section for direct command line output.
	- Added a "Linux" section detailing necessary dependencies for clipboard functionality, ensuring better usability for Linux users.

- **Bug Fixes**
	- Improved error handling and logging messages for invalid Git repository paths, enhancing user clarity during errors.

- **Refactor**
	- Standardized string delimiters for improved code consistency.
	- Enhanced parameterization of key functions to increase adaptability and usability.
	- Adjusted control flow to derive branch names more robustly from user inputs.

### CODE REVIEW

File: supporting_scripts/generate_code_cov_table/README.md [hunk 1]
```diff
@@ -89,4 +89,14 @@ The output will be separated into client and server sections, each containing a
...
 - Confirmation (assert/expect): A checkbox for manual confirmation of test coverage
 
+### Printing
+
+The output can be printed to the commandline using the `--print-results` option.
+
+### Linux
+
+On linux, pyperclip requires either `xclip` or `xsel` to be installed. 
+Alternativley, the python modules `qtpy` or `PyQT5` have to present.
```
Comment (JohannesStoehr) [inline supporting_scripts/generate_code_cov_table/README.md:99]:
```suggestion
Alternativley, the python modules `qtpy` or `PyQT5` have to be present. 
```

**Review by JohannesStoehr — CHANGES_REQUESTED** [review]

**Review by MaximilianAnzinger — APPROVED** [review]
Love the improvement ❤️

**Review by SimonEntholzer — APPROVED** [review]

**Review by coolchock — APPROVED** [review]

**Review by pzdr7 — APPROVED** [review]
Works well on Windows 👍

**Review by egekurt123 — APPROVED** [review]
LGTM

**Review by ole-ve — APPROVED** [review]
tested locally, works fine 👍

**Review by sarpsahinalp — APPROVED** [review]
Code lgtm

**Review by FelixTJDietrich — APPROVED** [review]
Code changes look good, thanks for the changes improving the code coverage script :)

**Review by krusche — APPROVED** [review]

**Pull request merged**